SYCL/ESIMD/regression/double_conversion_test.cpp

```diff
@@ -0,0 +1,82 @@
+//==--- double_conversion_test.cpp - DPC++ ESIMD on-device test ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This is basic test for testing conversion between double and unsigned int.
+
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <ios>
+#include <iostream>
+#include <typeinfo>
+
+namespace esimd = sycl::ext::intel::esimd;
+
+int test(sycl::queue queue, double testValue) {
+  double input = testValue;
+  unsigned int output = 0;
+  const unsigned int expected = static_cast<unsigned int>(
+      input); // exact values can be stored in double type, no
+              // implementation-defined rounding here
+
+  // Call simd conversion constructor
+  // Expectation: data not changed
+  {
+    sycl::range<1> range(1);
+    sycl::buffer<double, 1> buffer_in(&input, range);
+    sycl::buffer<unsigned int, 1> buffer_out(&output, range);
```

Comment:
Not a requirement: as soon as the test is 'double_conversion', it may be nice to test 'signed int' as well.